src/main/java/deterministicparrot/DeterministicParrot.java

```diff
@@ -0,0 +1,267 @@
+            throw new DeterministicParrotException("Invalid event format. Use /from and /to to specify event time.");
+        }
+        String eventName = String.join(" ", Arrays.copyOfRange(args, 1, fromIndex));
+        String startTime = String.join(" ", Arrays.copyOfRange(args, fromIndex + 1, toIndex));
+        String endTime = String.join(" ", Arrays.copyOfRange(args, toIndex + 1, args.length));
+        Event t = new Event(eventName, startTime, endTime);
+        addToList(t);
+    }
+
+    /**
+     * Deletes a task from the list and updates the UI.
+     *
+     * @param args The arguments passed to the delete command.
+     * @throws Exception If an error occurs while deleting the task.
+     */
+    private void deleteTask(String args[]) throws Exception {
+        if(args.length < 2){
+            throw new DeterministicParrotException("Please provide a task number.");
+        }
+        int i;
+        try {
+            i = Integer.parseInt(args[1]);
+        } catch (NumberFormatException e) {
+            throw new DeterministicParrotException("Please provide a valid task number.");
+        }
+        Task t = this.taskList.deleteTask(i);
+        this.ui.println("    " + "Noted. I've removed this task:");
+        this.ui.println("       " + t);
+        this.ui.println("     " + "Now you have " + this.taskList.getSize() + " tasks in the list.");
+        dumpTaskListToFile();
+    }
+    /**
+     * Displays the list of tasks in the UI.
+     */
+    private void printList(){
+        this.ui.println("     " + "Here are the tasks in your list:");
+        this.ui.println(this.taskList.formatAsString());
+    }
+    /**
+     * Performs necessary actions before exiting the application.
+     *
+     * @throws Exception If an error occurs during the exit process.
+     */
+
+    private void bye() throws Exception{
+        dumpTaskListToFile();
+        this.endParrot = true;
+        this.ui.bye();
+    }
+
+    /**
+     * Polls for user input and handles commands until the application is exited.
+     */
+    private void poll() {
+        this.ui.greet();
+        try{
+            this.taskList = TaskList.deserialize(storage.load());
+        } catch (Exception e) {
+            this.ui.println("     " + "No saved task list found. Starting with empty task list.");
+            this.taskList = new TaskList();
+        }
+
+        while (true) {
+            if(this.endParrot){
+                break;
+            }
+            String input = this.ui.readCommand();
+            if (input.isEmpty()) {
+                continue; // Skip empty input
+            }
+            this.ui.printDash();
+            try {
+                this.parser.handleCommand(input);
+            } catch (Exception e) {
+                this.ui.printError(e);
+            }
+            this.ui.printDash();
+        }
+    }
+
+    private void findTask(String args[]) throws Exception {
```

Comment:
Not sure if this method handles the case where there are no matching tasks. 
If it does, im sorry, im a coding noob :(
but wow, stringbuilder throws me back to CS2040S days 😨

Reply:
if there's no matching tasks i think it just doesn't print any of the tasks. will check again